### PR TITLE
[Actions] Pointagram

### LINKED ISSUE #3735: Pointagram
### App
Pointagram

### Triggers
* Find Player

### Actions
* Add Points
* Create Player
* Update Player Data
* Find Player

### PR DESCRIPTION
**Pull request by vellames**
Closes #3735 

Implemented:

- Add points to player
- Add player
- List players

Some actions were not implemented because the API does not provide the necessary endpoints

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/4131"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/pointagram/actions/add-points-to-player/add-points-to-player.mjs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import app from "../../pointagram.app.mjs";
+
+export default {
+  type: "action",
+  key: "pointagram-add-points-to-player",
+  name: "Add Points to Player",
+  description: "Add points to a player. [See the docs here](https://www.pointagram.com/custom-integration-gamification/)",
```
Comment (lcaresia) [inline components/pointagram/actions/add-points-to-player/add-points-to-player.mjs:7]:
```suggestion
  description: "Add points to a player. [See docs here](https://www.pointagram.com/custom-integration-gamification/)",
```

File: components/pointagram/actions/add-points-to-player/add-points-to-player.mjs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      description: "The number of points to add to the player",
+    },
+  },
+  async run({ $ }) {
+    const res = await this.app.addPointsToPlayer({
+      player_id: this.playerId,
+      points: this.points,
+      scoreseries_id: this.scoreSeriesId,
+    }, $);
+    $.export("$summary", "Points successfully added");
```
Comment (lcaresia) [inline components/pointagram/actions/add-points-to-player/add-points-to-player.mjs:35]: You can do this for the other exports too.

Obs: I din't tested this!

```suggestion
    $.export("$summary", `Points successfully added with id ${res.id}`);
```

File: components/pointagram/actions/add-points-to-player/add-points-to-player.mjs (hunk 1 above)
Comment (lcaresia) [inline components/pointagram/actions/add-points-to-player/add-points-to-player.mjs:7]: Fix this in other descriptions too!

```suggestion
  description: "Add points to a player. [See the docs here](https://www.pointagram.com/points-score-series/)",
```

File: components/pointagram/actions/list-players/list-players.mjs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  async run({ $ }) {
+    const params = {};
+    if (this.searchBy) {
+      params.search_by = this.searchBy;
+    }
+
+    if (this.searchValue) {
+      params.filter = this.searchValue;
+    }
+    const res = await this.app.listPlayers(params, $);
```
Comment (lcaresia) [inline components/pointagram/actions/list-players/list-players.mjs:35]:
```suggestion
    const {
      searchValue,
      searchBy,
    } = this

    const res = await this.app.listPlayers({
      search_by: searchBy,
      filter: searchValue,
    }, $);
```

File: components/pointagram/pointagram.app.mjs [hunk 4]
```diff
@@ -1,11 +1,105 @@
...
+      optional: true,
+    },
+    playerId: {
+      type: "string",
+      label: "Player ID",
+      description: "The ID of the player in Pointagram",
+      async options() {
+        const players = await this.listPlayers();
+        return players.map((player) => ({
+          label: player.Name,
```
Comment (lcaresia) [inline components/pointagram/pointagram.app.mjs:36]: The `Name` is correct?

```suggestion
          label: player.name,
```

File: components/pointagram/pointagram.app.mjs [hunk 5]
```diff
@@ -1,11 +1,105 @@
...
+      return params;
+    },
+    _getBaseUrl() {
+      return "https://app.pointagram.com/server";
+    },
+    _getHeaders() {
+      return {
+        "content-type": "application/json",
+        "api_key": this.$auth.api_key,
+        "api_user": this.$auth.api_user,
```
Comment (lcaresia) [inline components/pointagram/pointagram.app.mjs:70]: Move this to a getter method.

```suggestion
        "api_key": this._apiKey(),
        "api_user": this._apiUser(),
```

**Review by lcaresia — CHANGES_REQUESTED** [review]
Thanks for your contribution.

Just some fixes!

File: components/pointagram/actions/add-points-to-player/add-points-to-player.mjs (hunk 2 above)
Comment (vellames) [reply components/pointagram/actions/add-points-to-player/add-points-to-player.mjs:35]: The ids and the way these app works are pretty confusing. I don't think returning an id may help in something :)

File: components/pointagram/pointagram.app.mjs (hunk 4 above)
Comment (vellames) [reply components/pointagram/pointagram.app.mjs:36]: yeah

**Review by lcaresia — APPROVED** [review]
LGTM!

**Comment by DilanAthukorala** [comment]
Verification completed and ready for release.

**Pull request merged**